core/src/main/scala/org/apache/spark/status/api/v1/OneApplicationResource.scala

```diff
@@ -51,6 +52,21 @@ private[v1] class AbstractApplicationResource extends BaseAppResource {
   @Path("executors")
   def executorList(): Seq[ExecutorSummary] = withUI(_.store.executorList(true))
 
+  @GET
+  @Path("executors/{executorId}/threads")
+  def threadDump(@PathParam("executorId") executorId: String):
+  Option[Array[ThreadStackTrace]] = withUI { ui =>
```

Comment:
Not a big fan of exposing a private class (`ThreadStackTrace`) in a public API. (The api here is public even if the scala types aren't.)

Reply:
fixed, now it a response which able to store the error message too